Qiskit Upgrade and Deprecation Warnings

andre-a-alves opened the pull request:
<!--
⚠️ Your pull request title should be short, comprehensive, and understandable for all.
⚠️ If your pull request fixes an open issue, please link to the issue.
-->

<!--
If the validation checks fail
  1. Run `make check-types` (from the root directory of the repository) and fix any mypy (https://mypy.readthedocs.io/en/stable/) errors.

  2. Run `make check-style` and fix any flake8 (http://flake8.pycqa.org) errors.

  3. Run `make format` to format your code with the black (https://black.readthedocs.io/en/stable/index.html) autoformatter.

For more information, check the Mitiq style guidelines (https://mitiq.readthedocs.io/en/stable/contributing.html#style-guidelines).
-->

## Description

<!-- Please explain the changes you made here. -->
Closes #1835 and #1533

Imports `qiskit_aer` directly instead of importing and calling `qiskit.Aer`, and calls `AerSimulator` directly instead of calling `Aer.get_backend()`. These changes allow upgrading Qiskit to 0.43.0 and remove deprecation warnings.

---

### License

- [X] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

andreamari reviewed (approved):
Thanks @andre-a-alves, that was fast and effective! LGTM
@natestemen, I guess this is the first bounty of UnitaryHack.

natestemen commented on dev_requirements.txt:
Do we need to add `qiskit_aer` as a dependency here?

andre-a-alves replied:
At some point, but not yet.

Right now, Qiskit Aer is included as part of the Qiskit metapackage. There are plans to remove it in the future, but as long as Aer is still part of the metapackage, I think it makes sense to just keep them together.

natestemen replied:
Oh huh I don't think I knew you could have a package `X` that you pip install, and end with a situation where you can then `import X`, _and_ `import Y`. TIL.

andre-a-alves commented:
Yesterday while working on the docs, I realized they still used the old interface. I added a commit that corrects that.
I wasn't sure whether `IBMProvider` needed to be included int he dev requirements, since it's only necessary for the code in the docs, so I went ahead and added it, but I can easily remove it as well.

natestemen commented on docs/source/examples/cirq-ibmq-backends.md:
```diff
@@ -34,7 +34,7 @@ from mitiq import zne
 
 
-USE_REAL_HARDWARE = False
+USE_REAL_HARDWARE = True
```
I don't think we want the docs to try to use real hardware, do we?

natestemen replied:
Actually this is causing a failure on RTD: https://mitiq--1847.org.readthedocs.build/en/1847/examples/cirq-ibmq-backends.html

andre-a-alves replied:
Oh! Sorry about that. I had switched them to real hardware to test some of the changes and make sure IBMProvider worked. Latest merge fixes this.

Misty-W commented:
@natestemen, once @andre-a-alves resolves the conflicts, is this good to go?

natestemen reviewed (approved):
🚢 Thanks Andre!
